hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -0,0 +1,153 @@
+ * the License.
+ */
+
+package co.cask.plugin.etl.batch.commons;
+
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.hive.hcatalog.common.HCatException;
+import org.apache.hive.hcatalog.data.schema.HCatFieldSchema;
+import org.apache.hive.hcatalog.data.schema.HCatSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * <p>>Hive Schema Converter class to convert {@link Schema} to {@link HCatSchema} and vice versa.</p>
+ * Note: If the {@link HCatSchema} contains non-primitive type then this conversion to {@link Schema} will fail.
+ */
+public class HiveSchemaConverter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveSchemaConverter.class);
+
+  /**
+   * Converts a CDAP's {@link Schema} to Hive's {@link HCatSchema} while verifying the fields in the given
+   * {@link Schema} to exists in the table.
+   *
+   * @param schema the {@link Schema}
+   * @return {@link HCatSchema} for the given {@link Schema}
+   */
+  public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
+    List<HCatFieldSchema> fields = Lists.newArrayList();
+    for (Schema.Field field : schema.getFields()) {
+      String name = field.getName();
+      try {
+        // this field of the schema must exist in the table
+        Preconditions.checkNotNull(tableSchema.get(name), "Missing field %s in table schema", name);
+        fields.add(new HCatFieldSchema(name, getType(name, field.getSchema().getType()), ""));
+      } catch (HCatException e) {
+        LOG.error("Failed to create HCatFieldSchema field {} of type {} from schema", name,
+                  field.getSchema().getType());
+      }
+    }
+    return new HCatSchema(fields);
+  }
+
+  /**
+   * Returns the {@link PrimitiveTypeInfo} for the {@link Schema.Type}
+   *
+   * @param name name of the field
+   * @param category {@link Schema.Type} of the field
+   * @return {@link PrimitiveTypeInfo} for the given {@link Schema.Type} which is compatible with Hive.
+   */
+  private static PrimitiveTypeInfo getType(String name, Schema.Type category) {
+    switch (category) {
+      case BOOLEAN:
+        return TypeInfoFactory.booleanTypeInfo;
+      case INT:
+        return TypeInfoFactory.intTypeInfo;
+      case LONG:
+        return TypeInfoFactory.longTypeInfo;
+      case FLOAT:
+        return TypeInfoFactory.floatTypeInfo;
+      case DOUBLE:
+        return TypeInfoFactory.doubleTypeInfo;
+      case STRING:
+        return TypeInfoFactory.stringTypeInfo;
+      case BYTES:
+        return TypeInfoFactory.binaryTypeInfo;
+      default:
+        throw new IllegalArgumentException(String.format(
```

Comment:
this means nullable types are not supported (nullable is just a union of null and some other type).  I would think we would want to support nullable types.  You can pass in a Schema to this method instead of a Schema.Type, then use the isNullable() method and getNonNullable() at the start of the method.